code/handlers/ldap.q

```diff
@@ -50,29 +78,32 @@ login:{[user;pass]                                              / validate login
     incache[`time]>.z.p-.ldap.checktime;                        / previous attemp occured within checktime period
     incache[`pass]~np:md5 pass                                  / same password was used
   );
-    1b;
-    @[{.ldap.authenticate[x]`success};dict;0b]                  / attempt authentication
+    enlist[`ReturnCode]!enlist 0i;
+    .[.ldap.bind;(.ldap.sessionID;dict);0b]                 / attempt authentication
   ];
  
-  `.ldap.cache upsert (user;np;`$.ldap.server;.ldap.port;.z.p; (1+0^incache`attempts;0) authorised;authorised;0b);  / upsert details of current attempt
+  
+
+  `.ldap.cache upsert (user;np;`$.ldap.server;.ldap.port;.z.p; (1+0^incache`attempts;0) authorised[`ReturnCode] ;authorised[`ReturnCode]~0i;0b);  / upsert details of current attempt
 
-  $[authorised;                                                 / display authentication status message
+  $[authorised[`ReturnCode]~0i;                                                 / display authentication status message
     .ldap.out"successfully authenticated user ",;
-    .ldap.err"failed to authenticate user ",] dict`bind_dn;
+    .ldap.err"failed to authenticate user ",.ldap.err2string authorised[`ReturnCode]] dict[`bind_dn];
```

Comment:
I think there's a `,` missing here and some brackets (note that there's a trailing comma on each of the log lines to join the username onto the log message. Think the username is now `dn` instead of `bind_dn` as well

```suggestion
    .ldap.err"failed to authenticate user ",.ldap.err2string[authorised[`ReturnCode]],] dict[`dn];
```